src/cart/src/include/gurt/hash.h

```diff
@@ -523,20 +557,20 @@ void d_hash_table_debug(struct d_hash_table *htable);
  * set bit 0 to 1.
  */
 enum {
-	D_HTYPE_PTR		= 0, /**< pointer type handle */
+	D_HTYPE_PTR		= 0,	/**< pointer type handle */
 	/* Must enlarge D_HTYPE_BITS to add more types */
 };
 
 struct d_hlink;
 struct d_hlink_ops {
 	/** free callback */
-	void	(*hop_free)(struct d_hlink *rlink);
+	void	(*hop_free)(struct d_hlink *hlink);
 };
 
 struct d_rlink {
 	d_list_t		rl_link;
-	unsigned int		rl_ref;
-	unsigned int		rl_initialized:1;
+	uint32_t		rl_ref:30,
+				rl_initialized:1;
```

Comment:
This seems a little odd.  Using the :1 notation for a bool seems to predate compilers that support bool (we use it elsewhere in this header already) and the only other reason for using it is to pack numbers into a smaller space, but you seem to have a 30 bit int here and a 1 bit int.

Either you're packing into a smaller space in which case I'd expect padding or there's a specific requirement for rl_ref to be 30 bits specifically which I can't see anywhere and would be worthy of a comment.

Reply:
I had in mind less space usage by packing both members into single variable. But probably this is a bad idea. Especially if we will change this into atomic usage. Restored original.

Reply:
The issue here specifically is that this is part of the public API, so the cost of changing it in future is very high.  The correct type for rl_initialized would be bool.